packages/common/src/GRPCClient.ts

```diff
@@ -1413,6 +1416,22 @@ export class ConcordiumGRPCClient {
     private async getConsensusHeight() {
         return (await this.getConsensusStatus()).lastFinalizedBlockHeight;
     }
+
+    /**
+     * Queries the node to check its health
+     *
+     * {@codeblock ~~:client/healthCheck.ts#documentation-snippet}
+     *
+     * @returns a boolean indicating whether the node is healthy or not.
+     */
+    async healthCheck(): Promise<boolean> {
+        try {
+            await this.healthClient.check({});
+            return true;
+        } catch (e) {
+            return false;
+        }
+    }
```

Comment:
This doesn't seem right. On the GRPC types it is documented that `a return code of "OK" is used for success`. Would it not make more sense to check for this and propagate any GRPC errors thrown as well?

Reply:
Well it only returns if its "OK", otherwise it will throw. (So if its catching up it will also throw an error)
I've added the error message as part of the return value.

Reply:
Nice 😄